modelforge/potential/potential.py

```diff
@@ -435,6 +426,18 @@ def setup_potential(
     from modelforge.potential.utils import remove_units_from_dataset_statistics
     from modelforge.utils.misc import seed_random_number
 
+    if not dataset_statistic:  # set default value when value not passed
+        dataset_statistic = dict(
+            {
+                "training_dataset_statistics": {
+                    "per_atom_energy_mean": unit.Quantity(0.0, unit.kilojoule_per_mole),
+                    "per_atom_energy_stddev": unit.Quantity(
+                        1.0, unit.kilojoule_per_mole
+                    ),
+                }
+            },
+        )
+
```

Comment:
it might make sense to leave it in the signature of the function. it makes it very clear how a non-default dictionary should look like by presenting the default.

Reply:
I would suggest not using any `Dict` as a default value of a function as it may have some unexpected behaviors when you call the function multiple times or you reimport the module (reload the function). The problem is that the object is created when loading the function https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python.

Reply:
But we weren't passing an empty dictionary.  If `dataset_statistic` isn't passed, it will be initialized to the default values.  It's a bit confusing though, because these values can be overwritten based on the "use_default_dataset_statistic" flag, which if false by default, will read the dataset statistics from file (although I note there is no checking on whether this file exists).  I think this all needs a bit of a refactor in terms of the dataset_statistic  because there are way too many layers and branches.

Reply:
Mostly, I don't think this should keep us from merging the PR; these changes in the PR are overall straightforward and thus should be easy to merge, but this will impact other PRs downstream.